cla_backend/apps/call_centre/views.py

```diff
@@ -517,3 +527,71 @@ def get_queryset(self, *args, **kwargs):
         qs = super(CSVUploadViewSet, self).get_queryset(*args, **kwargs).filter(
             month__gte=after)
         return qs
+
+
+class DBExportView(APIView):
+    sql_files = {
+        'cases': 'ExportCases.sql',
+        'personal_details': 'ExportPersonalDetails.sql',
+        'standard': 'ExportStandardTables.sql',
+        'media_code_group': 'ExportMediaCodeGroup.sql',
+    }
+
+    authentication_classes = (OBIEESignatureAuthentication,)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    filename = 'cla_database.zip'
+
+    def get(self, request, format=None):
+        """
+        dt_from, dt_to: ISO 8601 datetime string (2014-08-29T23:59:59)
+        passphrase: diversity GPG private key passphrase
+        """
+        dt_from = request.QUERY_PARAMS.get('from', '')
+        dt_to = request.QUERY_PARAMS.get('to', '')
+        passphrase = request.QUERY_PARAMS.get('passphrase', '')
+
+        export_path = tempfile.mkdtemp()
```

Comment:
Yep, rate limit is a good idea - this method shouldn't be hit more than once a day anyway.